uniffi_core/src/ffi/rustbuffer.rs

```diff
@@ -228,6 +228,8 @@ pub fn uniffi_rustbuffer_from_bytes(
 ) -> RustBuffer {
     rust_call(call_status, || {
         let bytes = bytes.as_slice();
+        let mut bytes = bytes.to_vec();
+        bytes.shrink_to(i32::MAX as usize);
         Ok(RustBuffer::from_vec(bytes.to_vec()))
```

Comment:
This introduces at least one extra unnecessary copy because to_vec is now called twice. Did you actually see slice::to_vec producing Vec's with a capacity larger than i32::MAX? Or did you just change all places where RustBuffer::from_vec is called?

Reply:
Oops, yeah, no need for that extra `to_vec()`, thanks.

I have observed this issue happening in `Lower::lower_rust_into_vec` specifically. The patches in `rustbuffer` are only in the spirit of preventing a different code path from triggering the issue, but I'm happy to remove these if they are too heavy-handed.